boards/v2/qemu/qemu_arc/Kconfig

```diff
@@ -0,0 +1,23 @@
+# Copyright (c) 2020 Synopsys, Inc. All rights reserved.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+config BOARD_QEMU_ARC
+	select SOC_QEMU_ARC
+	select QEMU_TARGET
+	select HAS_COVERAGE_SUPPORT
+
+config BOARD_QEMU_ARC_QEMU_EM4
+	select SOC_QEMU_ARC_EM4
+
+config BOARD_QEMU_ARC_QEMU_HS3X
```

Comment:
same for this one, let's keep this as generic 'hs'.

Reply:
As above

Reply:
You can't have this both ways, the soc is hs3x and it was wrong to have hs in the original board then have hs3x_xip (which is a variant), you can either have both called hs (which in my opinion is wrong because there is no hs soc) or both named hs3x.

Reply:
The options here are as follows:

- The naming in this PR is picked, em4 and hs3x, to build for the hs3x you would use `qemu_arc/qemu_hs3x` and for XIP variant `qemu_arc/qemu_hs3x/xip`
- The naming in this PR is changed so that em4 falls back to em and hs3x falls back to hs, this means the hs3x_xip name will vanish and be hs
- The existing names from hwmv1 are used, this means the whole hs3x soc has to be duplicated as hs and hs3x, this then means hs and hs3x will be different boards in qemu, it means you will have `qemu_arc/qemu_hs`, `qemu_arc/qemu_hs3x` (which will fail to configure and throw an error, it will still appear in the list of supported board names) and `qemu_arc/qemu_hs3x/xip` which will be the XIP version

Reply:
I got mixed up between different socs it seems, it has been addressed